src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -634,31 +741,50 @@ private static VCFHeader warpHeader(VCFHeader in, Map<String, Long> namesAndLeng
    * then performs liftover on BED, then classifies regions
    * based on pre- and post- liftover regions.
    */
-  private static List<HomologousRange> processAndSaveBed(String inputBed, Fasta queryFasta,
+  private static List<HomologousRange> processAndSaveBed(String inputBed, String inputVcf,
+      Fasta queryFasta,
       Fasta targetFasta) {
     // Open necessary readers
     BufferedReader bedReader = null;
+    VCFFileReader vcfReader = null;
     try {
       logger.log(Level.INFO, "Reading BED");
       bedReader = Files.newBufferedReader(Paths.get(inputBed), UTF_8);
+      logger.log(Level.INFO, "Reading VCF");
+      vcfReader = new VCFFileReader(new File(inputVcf), false);
     } catch (IOException ex) {
       fail("failed to parse input BED/FASTA/VCF file(s): " + ex.getMessage());
     }
 
     // Takes the input BED and splits at non-DNA characters
     logger.log(Level.INFO, "Split DNA at non-DNA characters");
-    List<GenomeRange> dnaOnlyBED = null;
+    List<GenomeRange> dnaOnlyInputBED = null;
     try {
-      if ((dnaOnlyBED = splitAtNonDNA(queryFasta, bedReader)) == null) {
+      if ((dnaOnlyInputBED = splitAtNonDNA(queryFasta, bedReader)) == null) {
         fail("failed to generate reference genome");
       }
     } catch (IOException ex) {
       fail("failed to read from input bed: " + ex.getMessage());
     }
 
+    logger.log(Level.INFO, "Generating regions from variants");
+    List<GenomeRange> fromVcfBED = generateBEDFromVCF(vcfReader);
+
+    Collections.sort(fromVcfBED);
+    Collections.sort(dnaOnlyInputBED);
+
+    logger.log(Level.INFO, "Removing overlap from VCF ranges");
+    fromVcfBED = omitOverlap(fromVcfBED);
+
+    logger.log(Level.INFO, "Merging query regions with regions from VCF");
+    List<GenomeRange> mergedBed = mergeRegionsFromQueryBEDAndVariants(dnaOnlyInputBED, fromVcfBED);
+
+    logger.log(Level.INFO, String.format("Removing overlap in %d BED record(s)", mergedBed.size()));
+    List<GenomeRange> processedQueryBed = omitOverlap(mergedBed);
```

Comment:
If the above steps of splitting and merging are performed correctly, there should be zero overlapping records at this point.